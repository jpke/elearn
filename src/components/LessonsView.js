import React, {Component} from 'react';
import {Link} from 'react-router';

//view for LessonsContainer
//stateful component for lifecycle hook:
  //getLessons action call upon mounting
  //pulls down list of course pdf lessons
//allows user to select a lesson from list, which pulls preview/download url from server, then displays as a clickable "Preview" link
//clicking "Preview" will open pdf lesson in new browser tab
//unauthenticated user will be instructed to log in to view lessons
export default class LessonsView extends Component {
  constructor(props) {
    super(props);
    //only calls getLessons if user is authenticated (has jwt token)
    if(this.props.token) this.props.getLessons(this.props.token);
  }
  //getPDF action creates async request to server
  //receives lesson BOX preview/download url
  //requires lesson id and user token
  click(event) {
    this.props.getPDF(event.target.id, this.props.token);
  }

  render() {
    let lessons = this.props.lessons.map((lesson, index) => {
      return (
        <li key={index} id={lesson.id} onClick={this.click.bind(this)}>
          {lesson.name}
        </li>
      );
    });

    return (
      <div className="lessonsContainer">
        <h2>{this.props.courseName ?
              this.props.courseName + " Lessons"
              :
              "Lessons"}
        </h2>
        {this.props.token ?
          <div>
            <h3>pdfs</h3>
            <ul className="item-list">
              {lessons}
            </ul>
            {this.props.preview ?
              <div className="preview">
                <a href={this.props.preview} target="_blank">Preview</a>
              </div>
            : ""}
              {this.props.admin && <input type="file" id="uploadPDFInput" name="uploadPDF" onChange={() => {this.props.uploadPDF()}}/>}

          </div>
        :
          <h3>You must be <Link to='/' className="redirect">logged in</Link> to access this content</h3>}
      </div>
    );
  }
}
