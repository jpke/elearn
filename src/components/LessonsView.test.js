/* eslint-disable no-undef */
/* eslint-disable import/no-named-as-default */

import React from 'react';
import renderer from 'react-test-renderer';

import LessonsView from './LessonsView';

let lessons, preview;
beforeAll(() => {
  lessons = [
    {
      type: "file",
      id: "130861063664",
      etag: "1",
      name: "Example PDF.pdf",
      modified_at: "2017-02-02T12:29:25-08:00",
      size: 15342
    },
    {
      type: "file",
      id: "130861063665",
      etag: "1",
      name: "Example 2 PDF.pdf",
      modified_at: "2017-02-02T12:29:25-08:00",
      size: 15343
    }
  ];
  preview = {
    preview: "previewUrl",
    download: "fileDownloadUrl",
    name: "Example PDF.pdf",
    id: "130861063664",
  }
})

it('renders correctly for logged in non-admin user', () => {
  const tree = renderer.create(
    <LessonsView
      getLessons={jest.fn()}
      getPDF={jest.fn()}
      uploadPDF={jest.fn()}
      deletePDF={jest.fn()}
      token={"0000"}
      courseName={"example course"}
      courseID={"0000"}
      admin={false}
      lessons={lessons}
      loading={false}
      preview={preview}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders correctly for logged in admin user', () => {
  const tree = renderer.create(
    <LessonsView
      getLessons={jest.fn()}
      getPDF={jest.fn()}
      uploadPDF={jest.fn()}
      deletePDF={jest.fn()}
      token={"0000"}
      courseName={"example course"}
      courseID={"0000"}
      admin={true}
      lessons={lessons}
      loading={false}
      preview={preview}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})

it('renders correctly for logged out user', () => {
  const tree = renderer.create(
    <LessonsView
      getLessons={jest.fn()}
      getPDF={jest.fn()}
      uploadPDF={jest.fn()}
      deletePDF={jest.fn()}
      token={""}
      courseName={"example course"}
      courseID={"0000"}
      admin={false}
      lessons={lessons}
      loading={false}
      preview={preview}
    />
  ).toJSON();
  expect(tree).toMatchSnapshot();
})
