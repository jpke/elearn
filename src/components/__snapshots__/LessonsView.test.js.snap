exports[`test renders correctly for logged in admin user 1`] = `
<div
  className="lessonsContainer">
  <h2>
    example course Lessons
  </h2>
  <div>
    <h3>
      pdfs
    </h3>
    <ul
      className="item-list">
      <div>
        <li
          id="130861063664"
          onClick={[Function]}>
          Example PDF.pdf
        </li>
        <button
          onClick={[Function]}>
          Delete
        </button>
      </div>
      <div>
        <li
          id="130861063665"
          onClick={[Function]}>
          Example 2 PDF.pdf
        </li>
        <button
          onClick={[Function]}>
          Delete
        </button>
      </div>
    </ul>
    <div
      className="preview">
      <a
        href={
          Object {
            "download": "fileDownloadUrl",
            "id": "130861063664",
            "name": "Example PDF.pdf",
            "preview": "previewUrl",
          }
        }
        target="_blank">
        Preview
      </a>
    </div>
    <input
      id="uploadPDFInput"
      name="uploadPDF"
      onChange={[Function]}
      type="file" />
  </div>
</div>
`;

exports[`test renders correctly for logged in non-admin user 1`] = `
<div
  className="lessonsContainer">
  <h2>
    example course Lessons
  </h2>
  <div>
    <h3>
      pdfs
    </h3>
    <ul
      className="item-list">
      <div>
        <li
          id="130861063664"
          onClick={[Function]}>
          Example PDF.pdf
        </li>
      </div>
      <div>
        <li
          id="130861063665"
          onClick={[Function]}>
          Example 2 PDF.pdf
        </li>
      </div>
    </ul>
    <div
      className="preview">
      <a
        href={
          Object {
            "download": "fileDownloadUrl",
            "id": "130861063664",
            "name": "Example PDF.pdf",
            "preview": "previewUrl",
          }
        }
        target="_blank">
        Preview
      </a>
    </div>
  </div>
</div>
`;

exports[`test renders correctly for logged out user 1`] = `
<div
  className="lessonsContainer">
  <h2>
    example course Lessons
  </h2>
  <h3>
    You must be 
    <a
      className="redirect"
      onClick={[Function]}
      style={Object {}}>
      logged in
    </a>
     to access this content
  </h3>
</div>
`;
